third_party/move/move-compiler-v2/src/pipeline/reference_safety_processor.rs

```diff
@@ -1010,7 +1009,15 @@ impl<'env, 'state> LifetimeAnalysisStep<'env, 'state> {
     ///
     /// If we walk this graph now from the root to the leaves, we can determine safety by directly comparing
     /// hyper edge siblings.
-    fn check_borrow_safety(&mut self, temps: &BTreeSet<TempIndex>) {
+    fn check_borrow_safety(&mut self, temps_vec: &[TempIndex]) {
+        // First check direct duplicates
+        for (i, temp) in temps_vec.iter().enumerate() {
```

Comment:
You could probably do something like:
```
let temps = BTreeSet<_>::new();
for temp in temps_vec.iter().rev() {
    if temps.contains(temp) {
        self.exclusive_access_direct_dup_error(temp.clone())
   } else {
        temps.insert(temp.clone());
   }
}
```
and reduce this from `O(N^2)` to `O(N log N)` or so.

Or we can save that for later optimization if needed.

Reply:
Sorry I saw only after merge.

I'm certain there is a non-quadratic version, but I can't quite see what you mean,  this seems to generate an error for every iteration?

Reply:
No, only if we've already seen `temp` does it yield an error.  And we're processing `iter().rev()` so it's going backwards over the vec just as before.  Though actually, now that I think about it, that might not matter unless we are showing the location.